compiler/rustc_borrowck/src/diagnostics/mod.rs

```diff
@@ -623,6 +623,26 @@ impl UseSpans<'_> {
         }
     }
 
+    // Add a subdiagnostic to the use of the captured variable, if it exists.
+    pub(super) fn var_subdiag(
+        self,
+        err: &mut Diagnostic,
+        f: impl Fn(Span) -> crate::session_diagnostics::CaptureVarCause,
```

Comment:
I [bargained](https://github.com/rust-lang/rust/pull/103559#discussion_r1012805436) with the compiler